homeassistant/components/knx/schema.py

```diff
@@ -0,0 +1,309 @@
+"""Voluptuous schemas for the KNX integration."""
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_ADDRESS,
+    CONF_DEVICE_CLASS,
+    CONF_ENTITY_ID,
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PORT,
+    CONF_TYPE,
+)
+import homeassistant.helpers.config_validation as cv
+
+from .const import OPERATION_MODES, PRESET_MODES, ColorTempModes
+
+
+class ConnectionSchema:
+    """Voluptuous schema for KNX connection."""
+
+    CONF_KNX_LOCAL_IP = "local_ip"
+
+    TUNNELING_SCHEMA = vol.Schema(
+        {
+            vol.Required(CONF_HOST): cv.string,
+            vol.Optional(CONF_KNX_LOCAL_IP): cv.string,
+            vol.Optional(CONF_PORT): cv.port,
+        }
+    )
+
+    ROUTING_SCHEMA = vol.Schema({vol.Optional(CONF_KNX_LOCAL_IP): cv.string})
+
+
+class CoverSchema:
+    """Voluptuous schema for KNX covers."""
+
+    CONF_MOVE_LONG_ADDRESS = "move_long_address"
+    CONF_MOVE_SHORT_ADDRESS = "move_short_address"
+    CONF_POSITION_ADDRESS = "position_address"
+    CONF_POSITION_STATE_ADDRESS = "position_state_address"
+    CONF_ANGLE_ADDRESS = "angle_address"
+    CONF_ANGLE_STATE_ADDRESS = "angle_state_address"
+    CONF_TRAVELLING_TIME_DOWN = "travelling_time_down"
+    CONF_TRAVELLING_TIME_UP = "travelling_time_up"
+    CONF_INVERT_POSITION = "invert_position"
+    CONF_INVERT_ANGLE = "invert_angle"
+
+    DEFAULT_TRAVEL_TIME = 25
+    DEFAULT_NAME = "KNX Cover"
+
+    SCHEMA = vol.Schema(
+        {
+            vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+            vol.Optional(CONF_MOVE_LONG_ADDRESS): cv.string,
+            vol.Optional(CONF_MOVE_SHORT_ADDRESS): cv.string,
+            vol.Optional(CONF_POSITION_ADDRESS): cv.string,
+            vol.Optional(CONF_POSITION_STATE_ADDRESS): cv.string,
+            vol.Optional(CONF_ANGLE_ADDRESS): cv.string,
+            vol.Optional(CONF_ANGLE_STATE_ADDRESS): cv.string,
+            vol.Optional(
+                CONF_TRAVELLING_TIME_DOWN, default=DEFAULT_TRAVEL_TIME
+            ): cv.positive_int,
+            vol.Optional(
+                CONF_TRAVELLING_TIME_UP, default=DEFAULT_TRAVEL_TIME
+            ): cv.positive_int,
+            vol.Optional(CONF_INVERT_POSITION, default=False): cv.boolean,
+            vol.Optional(CONF_INVERT_ANGLE, default=False): cv.boolean,
+        }
+    )
+
+
+class BinarySensorSchema:
+    """Voluptuous schema for KNX binary sensors."""
+
+    CONF_STATE_ADDRESS = "state_address"
```

Comment:
Can we use constants for these class attributes so we can reuse constants between the schemas? Define the constants in a package module const.py.
```suggestion
    CONF_STATE_ADDRESS = CONF_STATE_ADDRESS
```

Reply:
Good idea, I took care of it.